core/arch/arm/kernel/generic_entry_a32.S

```diff
@@ -361,12 +361,24 @@ shadow_stack_access_ok:
 	bl	arm_cl2_enable
 #endif
 
+	/* Configure the MMU */
 	bl	core_init_mmu_map
 	bl	core_init_mmu_regs
+
+	/* Avoid losing any output due to the console_init call below */
+	bl	console_flush
+
+	/* Enable the MMU */
 	bl	cpu_mmu_enable
 	bl	cpu_mmu_enable_icache
 	bl	cpu_mmu_enable_dcache
 
+	/*
+	 * MMU is turned on: re-initialize the console so that the driver
+	 * uses the proper virtual addresses from now on
+	 */
+	bl	console_init
```

Comment:
Agreed. Would `console_init_early()` be OK? I have already started the work and realize now that I've swapped `init` and `early`, but both sound good to my french ear ;)